swift/ql/src/queries/Security/CWE-757/TLS13IsNotUsed.ql

```diff
@@ -0,0 +1,58 @@
+/**
+ * @name TODO
+ * @description TODO
+ * @kind path-problem
+ * @problem.severity error
+ * @security-severity TODO
+ * @precision high
+ * @id swift/tls13-is-not-used
+ * @tags security
+ *       external/cwe/cwe-757
+ */
+
+import swift
+import codeql.swift.dataflow.DataFlow
+import codeql.swift.dataflow.TaintTracking
+import codeql.swift.dataflow.FlowSources
+import DataFlow::PathGraph
+
+/**
+ * TODO
+ */
+class TLS13IsNotUsedConfig extends TaintTracking::Configuration {
+  TLS13IsNotUsedConfig() { this = "TLS13IsNotUsedConfig" }
+
+  override predicate isSink(DataFlow::Node node) {
+    exists(AssignExpr assign, MemberRefExpr member, string memberName |
+      assign.getSource() = node.asExpr() and
+      assign.getDest() = member and
+      memberName = member.getMember().(ConcreteVarDecl).getName() and
+      (
+        memberName = "tlsMinimumSupportedProtocolVersion" or
+        memberName = "tlsMinimumSupportedProtocol" or
+        memberName = "tlsMaximumSupportedProtocolVersion" or
+        memberName = "tlsMaximumSupportedProtocol"
+      )
+    )
+  }
+
+  override predicate isSource(DataFlow::Node node) {
+    exists(MethodRefExpr expr, EnumElementDecl enum, string enumName |
+      node.asExpr() = expr and
```

Comment:
Hmmm, wait. Is `tls_protocol_version_t.TLSv12` represented as a `MethodRefExpr` now @redsun82? Is that ... intentional?

Reply:
This one got me confused as well, I've been looking for `DeclRefExpr`s and was surprised to not see it there.
I'm glad we have PrintAst though 😅